docs/application/native/guides/machine-learning/machine-learning-training.md

```diff
@@ -434,6 +467,20 @@ Creating a dataset from a generator function is also possible.
                                                     NULL, NULL);
     ```
 
+3. Setting `user_data` for the dataset can be done with `ml_train_dataset_set_property()`, the `dataset` does not take the ownership of `user_data`, so be sure to free after use:
```

Comment:
While doing a re-review of the whole file again, I request you to kindly rephrase this such that the sentence is more clear. 

3. Setting `user_data` for the dataset can be done with `ml_train_dataset_set_property()`, the `dataset` does not take the ownership of `user_data`, so be sure to free after use:

What needs to be set free ?

Reply:
changed to ```so be sure to free `user_data` after use:```